src/Faker/Generator.php

```diff
@@ -205,6 +205,55 @@ class Generator
     protected $providers = [];
     protected $formatters = [];
 
+    /**
+     * @var UniqueGenerator
+     */
+    private $uniqueGenerator;
+
+    public function __construct()
+    {
+        $this->uniqueGenerator = new UniqueGenerator($this);
```

Comment:
Why?

Reply:
Each Generator instance has a `UniqueGenerator`. So I can run: 

```php
$faker->name(); // Might be 'Alice'
$faker->withUnique()->name(); // Might be 'Bob'
$faker->name(); // Might be 'Alice' again
$faker->withUnique()->name(); // Might be anything but 'Bob'
```

If we were about to create a new `UniqueGenerator` each time we called `withUnique()`, we could not guarantee it's uniqueness. Unless we used static variables, but then the uniqueness would be global and not specific to a instance of the `Generator`.

Reply:
Instead of reimplementing the behaviour here, what do you think about moving methods from `Faker\Provider\Base` to `Faker\Generator` one by one, guided by tests?

Also see https://github.com/FakerPHP/Faker/pull/233.